### DISCUSSION TITLE
MVVM模式下，异步更新UI问题

### DISCUSSION THREAD

**Discussion by minChaojie** [BYJRK/DotNet-Discussions#173] [Q&A]
我把车辆数据的接收和处理都放在一个异步方法中处理，处理结束后以Message的方式发布出来，展示界面的ViewModel订阅这个消息；
但是在更新VM中的属性值的时候报错，需要在主线程上修改，有什么解决办法吗？

**Comment by ErickWei** [comment]
这个问题我也问过，貌似没有其他办法，判断是不是主线程，如果不是，就 Dispatcher。
如果大神们有其他方法，求指教。感觉在现实开发中有很多这种需求


``` C#
if (Thread.CurrentThread != Application.Current.Dispatcher.Thread)
  {
        Application.Current.Dispatcher.Invoke(() =>
          {
                    /// your code
          });
   }
   else
   {
               /// your code
   }

```

**Comment by ErickWei** [comment]
[参考这个回复](https://github.com/BYJRK/DotNet-Discussions/discussions/55#discussioncomment-5318374)

**Comment by CodingOctocat** [comment]
你是否绑定一个 `ObservableCollection` 属性，然后对这个集合进行添加或删除吗？如果可以的话，考虑直接 new 一个新的集合给这个属性。或者尝试:

```cs
private readonly object _syncLock = new();
BindingOperations.EnableCollectionSynchronization(ObservableCollection, _syncLock);
```
